setuptools/tests/test_bdist_egg.py

```diff
@@ -47,7 +47,7 @@ def test_bdist_egg(self, setup_context, user_override):
         assert re.match(r'foo-0.0.0-py[23].\d+.egg$', content)
 
     @pytest.mark.xfail(
-        os.environ.get('PYTHONDONTWRITEBYTECODE'),
+        os.environ.get('PYTHONDONTWRITEBYTECODE', False),
```

Comment:
I avoid adding literals where they're not needed. Let's not change the semantics. Just disable mypy if it's too picky.

Reply:
I found an open pytest issue for this: https://github.com/pytest-dev/pytest/issues/10094

Reply:
PR opened: https://github.com/pytest-dev/pytest/pull/11565

Reply:
Pytest's api doesn't contractually support None (it works because it's falsy). See: https://github.com/pytest-dev/pytest/pull/11600

Which means this will always be an error (true positive) in static type-checkers (meaning it'll always need a `# type: ignore` comment), or we can be explicit and say that when the environment variable `PYTHONDONTWRITEBYTECODE` doesn't exist, it's equivalent to passing `False` to the condition for setuptools. (which is what that change was doing)

Reply:
I don't feel strongly about it. Happy with whatever resolution you chose.